addons/misra.py

```diff
@@ -1628,6 +1638,35 @@ def misra_18_5(self, data):
                 self.reportError(var.nameToken, 18, 5)
 
 
+    def misra_18_7(self, data):
+        structScopes = []
+        for token in data.tokenlist:
+            if not isStruct(token):
+                continue
+            structScopes.append(token.valueType.typeScope)
+
+        for scope in structScopes:
```

Comment:
I suggest that you remove the `structScopes` above and change this loop:

    for scope in data.scopes:
        if scope.type != 'Struct':
            continue